mysk-data-api/src/routes/v1/subjects/electives/enroll_electives.rs

```diff
@@ -65,44 +70,54 @@ pub async fn enroll_elective_subject(
     if !DbElectiveSubject::is_enrollment_period(pool).await? {
         return Err(Error::InvalidPermission(
             "The elective's enrollment period has ended".to_string(),
-            format!("/subjects/electives/{session_code}/enroll"),
+            format!("/subjects/electives/{elective_subject_session_id}/enroll"),
         ));
     }
 
     // Checks if the student is in a class available for the elective
-    if !DbElectiveSubject::is_student_eligible(pool, session_code, student_id).await? {
+    if !DbElectiveSubject::is_student_eligible(pool, elective_subject_session_id, student_id)
+        .await?
+    {
         return Err(Error::InvalidPermission(
             "Student is not eligible to enroll in this elective".to_string(),
-            format!("/subjects/electives/{session_code}/enroll"),
+            format!("/subjects/electives/{elective_subject_session_id}/enroll"),
         ));
     }
 
     // Checks if the student has already enrolled in the elective before
+    let subject_id = DbElectiveSubject::get_by_id(pool, elective_subject_session_id)
+        .await?
+        .subject_id;
+
     let enroll_count = query!(
         r"
-        SELECT COUNT(*) FROM student_elective_subjects
-        WHERE student_id = $1 AND elective_subject_id = $2
+        SELECT 
+            COUNT(*) 
+        FROM 
+            elective_subject_session_enrolled_students 
+            INNER JOIN elective_subject_sessions_with_detail_view on elective_subject_session_enrolled_students.elective_subject_session_id = elective_subject_sessions_with_detail_view.id
+        WHERE student_id = $1 AND subject_id = $2
```

Comment:
Why are you selecting from `elective_subject_session_enrolled_students` and then also joining with `elective_subject_sessions_with_detail_view`, which then selects from `elective_subject_session_enrolled_students` again...

You could probably optimise it to:
```sql
SELECT
    COUNT(*)
FROM
    elective_subject_session_enrolled_students enrolls
    INNER JOIN elective_subject_sessions electives
    ON enrolls.elective_subject_session_id = electives.id
WHERE student_id = $1 AND subject_id = $2
```

Reply:
I forgot to add `year` and `semester` filter :(